content/_releases/23/08.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Product Update - v23.08
+description: Details of the product update v23.08 from 27th of February 2023.
+layout: rel
+version: "23.08"
+section: Product Updates
+category: product-updates
+releaseDate: 2023-02-27
+redirect_from:
+  - /releases/23.08.html
+---
+
+## Features
+
+### Configurable Recipe Use for Contracts
+
+As a continuation of the updates introduced in Release [23.06](/releases/23/06#new-recipe-createedit-ui-features), we have added new features in recipe management. This feature allows you to control which users are allowed to create/edit Recipes (by default, all users can use the recipes feature). You can set it up with a new permission `tenants.contract.edit_feature_flags` - allow to set the feature flags for Contract.
+
+In this regard, POST and PATCH v2/contracts endpoins were changed - added an ability to set the feature flags for Contract.
+
+> **Please Note:** this permission belongs to tenant admins by default.
+
+Herer you can see an example of a request body:
+
+```
+{
+  "data": {
+    "type": "contract",
+    "attributes": {
+      "name": "Contract name",
+      "available_roles": [
+        {
+          "scope": "contracts",
+          "role": "admin"
+        }
+      ],
+      "flow_stats_enabled_default": true,
+      "support_user_id": "5b4f337bff4304610483ba67",
+      "custom_data": {
+        "description": "Contract description"
+      },
+      "feature_flags": {
+        "disable_recipe_feature": true
+      }
+    }
+  }
+}
+```
+
+> **Please Note:** "disable_recipe_feature" set to "true" be default.
+
+### Extend the UI view class `RESTAPIView` with the `application/x-ndjson`
+
+We transformed "Content type" dropdown on the "Body" tab of the REST API v2 component to a text field with a dropdown of a hints:
+
+{% include img.html max-width="60%" url="/assets/img/RN/2308/rn2308-rest-api-v2.png" title="Export Recipe" %}
+
+- the hints appear on focus into the field and list changes when you're entering something into the text field (search for existing values)
+- you can choose an item from the hints dropdown
+- if nothing to choose, you can type your custom content type and save it
+- we validate Content-type header according to [list of all the official MIME types](https://www.iana.org/assignments/media-types/media-types.xhtml).
+
+### Reducing the refresh time of auth-secrets
+
+Before we refresh token for auth-secrets in 1 minute before exparation. But some external APIs (as example, [ExactOnline](https://support.exactonline.com/community/login?ec=301&startURL=%2Fcommunity%2Fs%2Fknowledge-base%23All-All-DNO-Simulation-gen-apilimits)) don't allow to refresh secret very early.
```

Comment:
```suggestion
We now refresh auth-secret tokens closer to the token expiry. This change to 30 seconds before expiry (it used to be 60 seconds) is because various external APIs do not allow refreshes 1 minute before expiry (as example, [ExactOnline](https://support.exactonline.com/community/login?ec=301&startURL=%2Fcommunity%2Fs%2Fknowledge-base%23All-All-DNO-Simulation-gen-apilimits)) don't allow to refresh secret very early.
```